data-reconciliation-app/src/models/summary-record.ts

```diff
@@ -0,0 +1,145 @@
+import { MINIMUM_VOTES_THRESHOLD } from "../utils/constants";
+import { ServiceResult } from "../utils/service-result";
+import { DataAttributeType } from "./data-record";
+import { ReconciledRecord } from "./reconciled-record";
+
+export enum SummaryGroupStatus {
+  NotEnoughVotes = "NOT_ENOUGH_VOTES",
+  LackOfConsensus = "LACK_OF_CONSENSUS",
+  InConsensus = "IN_CONSENSUS",
+}
+
+export enum SummaryStatus {
+  Majority = "Majority",
+  Minority = "Minority",
+}
+
+export interface SummaryRecordProps {
+  key: string;
+  value: DataAttributeType;
+  type: string;
+  status: SummaryStatus;
+  groupStatus: SummaryGroupStatus;
+  membersInAgreementCount: number;
+  membersInDisagreementCount: number;
+  votesCount: number;
+  uniqueValuesCount: number;
+}
+
+export class SummaryRecord implements SummaryRecordProps {
+  readonly key: string;
+  readonly value: DataAttributeType;
+  readonly type: string;
+  readonly status: SummaryStatus;
+  readonly groupStatus: SummaryGroupStatus;
+  readonly membersInAgreementCount: number;
+  readonly membersInDisagreementCount: number;
+  readonly activeMembersCount: number;
+  readonly votesCount: number;
+  readonly uniqueValuesCount: number;
+
+  private constructor(summaryRecord: SummaryRecordProps) {
+    this.key = summaryRecord.key;
+    this.value = summaryRecord.value;
+    this.type = summaryRecord.type;
+    this.status = summaryRecord.status;
```

Comment:
nit naming this status and also having a groupStatus is confusing to me...however, I am struggling to come up with a better name myself...

minorityMajority? Open to other suggestions

Reply:
Done